### ISSUE TITLE
Global Imports in cosmos config aren't resolving correctly

### ISSUE THREAD

**Issue by SamStonehouse** [react-cosmos/react-cosmos#1332]
### What's wrong?

After adding global imports to my cosmos config I get module resolution errors
```
ERROR in C:codewebpack-5-yarninclude
Module build failed: UnhandledSchemeError: Reading from "C:codewebpack-5-yarninclude" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "c:" URIs.
    at C:\code\webpack-5-yarn\.yarn\$$virtual\webpack-virtual-05dfa2142e\0\cache\webpack-npm-5.39.1-99b3815143-1f8d2bd545.zip\node_modules\webpack\lib\NormalModule.js:767:26
```

### Steps to reproduce

I'm using a yarn 2 monorepo with pnp, the pnp is setup in my webpack config as shown in the pnp example project, however the monorepo makes the rest of the setup a bit different. I've added cosmos to the project root as I failed to get it working as it's own package.

Trimmed down reconstructed webpack config:

```
module.exports = {
  target: 'web',
  entry: 'C:\\code\\webpack-5-yarn\\entry.tsx',
  output: {
    path: 'C:\\code\\webpack-5-yarn\\dist\\public',
    filename: 'assets/bundle.js'
  },
  module: {
    rules: [
      {
        test: /\.(j|t)sx?$/,
        exclude: /node_modules/,
        use: { loader: 'babel-loader', options: { root: undefined } }
      }
    ]
  },
  plugins: [],
  resolve: {
    extensions: [ '.ts', '.js', '.tsx', '.jsx' ],
    plugins: [PnpWebpackPlugin]
  },
  resolveLoader: { plugins: [ PnpWebpackPlugin.moduleLoader(module) ] },
  mode: 'production'
}
```

Cosmos config

```
{
  "$schema": "http://json.schemastore.org/cosmos-config",
  "webpack": {
    "configPath": "./cosmos/webpack.cosmos.js"
  },
  "rootDir": "..",
  "globalImports": [
    "./include"
  ]
}
```

I have tried an absolute path to the script I want to include, a local file which then imports the script I want to include and just directly a script I want to include (which in this case is `@workspace-example/styles` which is an internal package of this monorepo

### User info

Relevent module versions:
    "react-cosmos": "^5.6.2",
    "webpack": "^5.39.1",
    "react": "^17.0.2",
    "pnp-webpack-plugin": "^1.6.4",

I don't really understand what's going on, it's not an error I've ever come across before. I have used cosmos 5 in the same way in a mono-repo managed by yarn before but this was yarn 1 without pnp

**Comment by SamStonehouse** [comment]
I've created another repository without yarn 2 pnp and workspaces, still having the same issue, you can see the repo here:
https://github.com/SamStonehouse/webpack-typescript-react-example

The error in this project is mostly the same:
```
ERROR in C:codeexample-projectsourcestylescommon.scss
Module build failed: UnhandledSchemeError: Reading from "C:codeexample-projectsourcestylescommon.scss" is not handled by plugins (Unhandled scheme).
Webpack supports "data:" and "file:" URIs by default.
You may need an additional plugin to handle "c:" URIs.
    at C:\code\example-project\node_modules\webpack\lib\NormalModule.js:767:26
    at Hook.eval [as callAsync] (eval at create (C:\code\example-project\node_modules\tapable\lib\HookCodeFactory.js:33:10), <anonymous>:6:1)
    at Hook.CALL_ASYNC_DELEGATE [as _callAsync] (C:\code\example-project\node_modules\tapable\lib\Hook.js:18:14)
    at Object.processResource (C:\code\example-project\node_modules\webpack\lib\NormalModule.js:764:9)
    at processResource (C:\code\example-project\node_modules\loader-runner\lib\LoaderRunner.js:220:11)
    at iteratePitchingLoaders (C:\code\example-project\node_modules\loader-runner\lib\LoaderRunner.js:171:10)
    at runLoaders (C:\code\example-project\node_modules\loader-runner\lib\LoaderRunner.js:397:2)
    at NormalModule.doBuild (C:\code\example-project\node_modules\webpack\lib\NormalModule.js:754:3)
    at NormalModule.build (C:\code\example-project\node_modules\webpack\lib\NormalModule.js:908:15)
    at C:\code\example-project\node_modules\webpack\lib\Compilation.js:1318:12
 @ ./node_modules/react-cosmos/dist/plugins/webpack/client/userDeps.js 6:0-59
 @ ./node_modules/react-cosmos/dist/plugins/webpack/client/index.js

webpack compiled with 1 error and 3 warnings
```

**Comment by skidding** [comment]
Hi @SamStonehouse,

Based on the fact that the error contains this path `C:codeexample-projectsourcestylescommon.scss` and that the repository you referenced works just fine on my macOS machine, I suspect this has to do module path resolution on Windows. We've addressed this sort of issues in the past but either some old issue crept back into the codebase or some of your plugins are doing something fishy.

It's difficult to debug this as I don't own a Windows machine and as I said, the repository works well on my end, but here's something that could gives us a clue.

Put this code in a file at `./cosmos/print-cosmos-config.js`

```js
const { detectCosmosConfig } = require('react-cosmos');
console.log(detectCosmosConfig());
```

Then run `node ./print-cosmos.config.js` and let us know what's the output. I'm particularly curious of what shows up under `globalImports`, as it will indicate if the path is already invalid at this point or the slashes get removed later down the line.

**Comment by SamStonehouse** [comment]
This is what it looks like
```
$ node print-cosmos-config.js 
{
  '$schema': 'http://json.schemastore.org/cosmos-config',
  webpack: { configPath: './cosmos/webpack.cosmos.js' },
  rootDir: 'C:\\code\\example-project',
  globalImports: [ 'C:\\code\\example-project\\source\\styles\\common.scss' ],
  exportPath: 'C:\\code\\example-project\\cosmos-export',
  disablePlugins: true,
  fixtureFileSuffix: 'fixture',
  fixturesDir: '__fixtures__',
  hostname: null,
  https: false,
  httpsOptions: null,
  port: 5000,
  publicUrl: '/',
  staticPath: null,
  userDepsFilePath: 'C:\\code\\example-project\\cosmos.userdeps.js',
  watchDirs: [ 'C:\\code\\example-project' ],
  experimentalRendererUrl: null,
  ui: {}
}
```
With my cosmos config as
```
{
  "$schema": "http://json.schemastore.org/cosmos-config",
  "webpack": {
    "configPath": "./cosmos/webpack.cosmos.js"
  },
  "rootDir": "..",
  "globalImports": ["source/styles/common.scss"]
}
```

Some extra info:
Node version 14.17.1
Running cosmos through both yarn and npm has the same ouput.

I'm going to go through the plugins now and see if removing them helps

**Comment by SamStonehouse** [comment]
Removing all plugins except HtmlWebpackPlugin (which I think is required?) still produces the error, approx webpack.js for reference

```
{
  target: 'web',
  entry: 'C:\\code\\example-project\\entry.tsx',
  output: {
    path: 'C:\\code\\example-project\\dist\\public',
    filename: 'assets/bundle.js'
  },
  plugins: [
    HtmlWebpackPlugin {
      userOptions: {
        template: 'C:\\code\\example-project\\cosmos\\index.html',
        filename: 'index.html'
      },
      version: 5
    }
  ],
  resolve: { extensions: [ '.ts', '.js', '.tsx', '.jsx' ] },
  module: {
    rules: [
      {
        test: /\.(j|t)sx?$/,
        exclude: /node_modules/,
        use: { loader: 'babel-loader', options: { root: undefined } }
      }
    ]
  },
  mode: 'development'
}
```

I asked somebody else to run it and they get the same error (on Windows) so at least it's not just me. I appreciate the time at least, I'll try and dig into it a bit :)

**Comment by skidding** [comment]
One quick idea: Try replacing the scss module in globalImports with a plain js module to clarify that the problem is path-related and not SASS complication related.

**Comment by SamStonehouse** [comment]
I tried that, no dice unfortunatly, but I think I've found the source of the issue -maybe-

userDepsTemplate.js generates this file (I pulled this out of the debugger so I think the \\ are just escaped \):
```
// This file is automatically generated by Cosmos. Add it to .gitignore and
// only edit if you know what you're doing.

// Keeping global imports here is superior to making them bundle entry points
// because this way they become hot-reloadable.
import 'C:\\code\\example-project\\source\\styles\\common.scss';

import fixture0 from 'C:/code/example-project/source/example-component/__fixtures__/example-component.tsx';
import fixture1 from 'C:/code/example-project/source/example-component/example-component.fixture.tsx';



export const rendererConfig = {
  "containerQuerySelector": null
};

export const fixtures = {
  'source/example-component/__fixtures__/example-component.tsx': fixture0,
  'source/example-component/example-component.fixture.tsx': fixture1
};

export const decorators = {

};
```

By changing the generation to this it works:
```
// This file is automatically generated by Cosmos. Add it to .gitignore and
// only edit if you know what you're doing.

// Keeping global imports here is superior to making them bundle entry points
// because this way they become hot-reloadable.
import 'C:/code/example-project/source/styles/common.scss';

import fixture0 from 'C:/code/example-project/source/example-component/__fixtures__/example-component.tsx';
import fixture1 from 'C:/code/example-project/source/example-component/example-component.fixture.tsx';



export const rendererConfig = {
  "containerQuerySelector": null
};

export const fixtures = {
  'source/example-component/__fixtures__/example-component.tsx': fixture0,
  'source/example-component/example-component.fixture.tsx': fixture1
};

export const decorators = {

};
```

I did this by changing

```
function resolveImportPath(filePath, relativeToDir) {
  return relativeToDir ? (0, _slash.default)(".".concat(_path.default.sep).concat(_path.default.relative(relativeToDir, filePath))) : filePath;
```
to
```
function resolveImportPath(filePath, relativeToDir) {
  return relativeToDir ? (0, _slash.default)(".".concat(_path.default.sep).concat(_path.default.relative(relativeToDir, filePath))) : filePath.replace(/\\/g, '/');
}
```

in `userDepsTemplate.js`

Not really sure what an actual fix would be as there may be a reason for using the windows directory format, otherwise I think maybe changing the global imports part of the template generation to just change \ to / might even be enough:

```
  .concat(globalImports.map(function (p) {
    return "import '".concat(resolveImportPath(p, relativeToDir).replace(/\\/g, '/'), "';");
  })
```

It does feel kinda bad and I've no idea if this would break anything else, any ideas on a better way to change it?

**Comment by skidding** [comment]
@SamStonehouse Thanks for the contribution! Your patch seemed safe enough so I published `react-cosmos@5.6.3` with it. Please try it out and let me know if it works as expected.

**Comment by SamStonehouse** [comment]
It's working great, thank you very much :)
